python/amici/petab_import.py

```diff
@@ -20,42 +21,51 @@
 logger = logging.getLogger(__name__)
 
 
-def get_fixed_parameters(condition_df: pd.DataFrame,
-                         sbml_model: 'libsbml.Model',
-                         const_species_to_parameters: bool = False):
+def get_fixed_parameters(
+        sbml_model: 'libsbml.Model',
+        condition_df: Optional[pd.DataFrame] = None,
+        const_species_to_parameters: bool = False) -> List[str]:
     """Determine, set and return fixed model parameters
 
     Parameters specified in `condition_file_name` are turned into constants.
     Only global SBML parameters are considered. Local parameters are ignored.
 
     Arguments:
         condition_df:
-            PEtab condition table as pandas.dataframe
+            PEtab condition table. If provided, the respective parameters
+            will be turned into AMICI constant parameters.
         sbml_model:
             libsbml.Model instance
         const_species_to_parameters:
             If `True`, species which are marked constant within the SBML model
             will be turned into constant parameters *within* the given
             `sbml_model`.
+
+    Returns:
+        List of IDs of parameter which are to be considered constant
     """
 
     # column names are model parameter names that should be made constant
     # except for any overridden parameters
     # (Could potentially still be made constant, but leaving them might
     # increase model reusability)
-    fixed_parameters = list(condition_df.columns)
-    try:
-        fixed_parameters.remove('conditionName')
-    except ValueError:
-        pass
-    # remove overridden parameters
-    fixed_parameters = [p for p in fixed_parameters
-                        if condition_df[p].dtype != 'O']
-    # must be unique
-    assert(len(fixed_parameters) == len(set(fixed_parameters)))
+    if condition_df is not None:
+        fixed_parameters = list(condition_df.columns)
+        try:
+            fixed_parameters.remove('conditionName')
+        except ValueError:
+            pass
+        # remove overridden parameters
+        fixed_parameters = [p for p in fixed_parameters
+                            if condition_df[p].dtype != 'O']
+        # must be unique
+        assert(len(fixed_parameters) == len(set(fixed_parameters)))
```

Comment:
note that assert statements are deleted in python optimized execution. thus it may be preferable to use explicit checks with raises.

Reply:
Right.